test/integration/api/services_controller_test.rb

```diff
@@ -90,6 +90,35 @@ class SettingsTest < self
       assert hosted_option.attribute('checked').present?
     end
 
+    test 'shows authentication options' do
+      # Default value
+      service.stubs(:backend_version).returns(nil)
+      get settings_admin_service_path(service)
+
+      page = Nokogiri::HTML::Document.parse(response.body)
+      default = page.at_css('#service_proxy_authentication_method_1')
+      assert default.attribute('checked').present?
+      service.unstub(:backend_version)
+
+      # API Key (user_key)
+      service.update!(backend_version: '1')
+      get settings_admin_service_path(service)
+
+      page = Nokogiri::HTML::Document.parse(response.body)
+      api_key_option = page.at_css('#service_proxy_authentication_method_1')
+      assert api_key_option.attribute('checked').present?
+      service.unstub(:backend_version)
```

Comment:
This `unstub` seems to be redundant because that is not `stubbed` beforehand. Do you need stubbing in this test case?

Reply:
Yes, since I do 

https://github.com/3scale/porta/blob/b96010ed6062d18d4541dd58de4700f434188c91/test/integration/api/services_controller_test.rb#L95

If I don't have that `unstab` the test fails. 

On the other hand, though it appears with the `returns(nil)`, I'm deleting it.

Reply:
But on line 101 you already unstubbed.

Reply:
Also what happens if you don't stub that in the first place?